### ISSUE TITLE
The integer versions of the `1/x` node should not exist

### ISSUE THREAD

**Issue by iamgreaser** [Neos-Metaverse/NeosPublic#3581]
### Describe the bug?

The `1/x` node has integer variants of it. Not only should these not exist, but they also divide by zero when fed a `0` input.

The exact behaviour is as follows:

- `1` -> `1`
- `-1` -> `-1`
- `0` -> RED NODE
- Anything else -> `0`

The bug is that these variants not only exist, but it is also easy to end up with them when one doesn't want them.

### Relevant issues

A search yielded no results.

### To Reproduce

Create a `1/x` LogiX node. Create an `int` input. Feed it into the `1/x` node. This will give you an integer `1/x` node.

This works for all vector variants, for the int (32-bit) and long (64-bit) families, and for both signed and unsigned variants. Narrower integer types are cast to their 32-bit counterparts.

### Expected behavior

When fed an integer input, a `1/x` node should convert the integer to the appropriate floating-point type before calculating the reciprocal. There should be no integer `1/x` nodes, as these make no sense.

Furthermore, internally, the behaviour of the node should be identical to feeding `1` and the input into a divide node and taking the output.

For backwards-compatibility, just in case someone actually somehow relied on any of these nodes, they may be converted to use a divide node with a numerator of `1`. This will behave identically, except that dividing by `0` will give a sane fallback value instead of red-noding.

### Screenshots

The nodes are, going from left-to-right, subgrouped top-to-bottom, keeping vector variants grouped:

- Color (RGBA)
- float, double, decimal
- int, uint
- long, ulong

I am not aware of any other variants of `1/x` which exist. I tried everything which wasn't an enum under the `Input` section, and tried one enum just in case.

![image](https://user-images.githubusercontent.com/138113/152630547-81d45ee6-1b37-4b7a-9ba0-d32cc2902de5.png)

### How often does it happen?

Always

### Does the bug persist after restarting Neos?

Yes

### Neos Version Number

2022.1.28.1335

### What Platforms does this occur on?

Windows, Linux

### Did this work before?

I Don't Know

### Additional context

I discovered this while trying to find an elegant way to make Neos not try to collapse an int-to-float cast which I was feeding into a divide node.

I noticed this was wrong the moment I saw the input and output to the node turn green.

### Reporters

GreaseMonkey.

**Comment by epicEaston197** [comment]
yes this does happen I was able to recreate it with the instructions provided
![image](https://user-images.githubusercontent.com/76523692/152641664-46eb942a-7a3f-457f-a1e1-655d0d3839bf.png)
